packages/app/src/cli/models/extensions/specifications/payments_app_extension_schemas/credit_card_payments_app_extension_schema.ts

```diff
@@ -0,0 +1,84 @@
+
+export const CREDIT_CARD_TARGET = 'payments.credit-card.render'
+export const CreditCardPaymentsAppExtensionSchema = BaseSchema.extend({
+  targeting: zod.array(zod.object({target: zod.literal(CREDIT_CARD_TARGET)})).length(1),
+  api_version: zod.string(),
+  payment_session_url: zod.string().url(),
+  refund_session_url: zod.string().url(),
+  capture_session_url: zod.string().url(),
+  void_session_url: zod.string().url(),
+  verification_session_url: zod.string().url().optional(),
+  confirmation_callback_url: zod.string().url().optional(),
+  supports_3ds: zod.boolean(),
+  supported_countries: zod.array(zod.string()),
+  supported_payment_methods: zod.array(zod.string()),
+  supports_installments: zod.boolean(),
+  supports_deferred_payments: zod.boolean(),
+  test_mode_available: zod.boolean(),
+  merchant_label: zod.string().max(MAX_LABEL_SIZE),
+  buyer_label: zod.string().max(MAX_LABEL_SIZE).optional(),
+  buyer_label_translations: zod.array(zod.object({locale: zod.string(), label: zod.string()})).optional(),
+  encryption_certificate: zod.object({
+    fingerprint: zod.string(),
+    certificate: zod.string().regex(CERTIFICATE_REGEX),
+  }),
+  checkout_payment_method_fields: zod
+    .array(
+      zod.object({
+        type: zod.union([zod.literal('string'), zod.literal('number'), zod.literal('boolean')]),
+        required: zod.boolean(),
+        key: zod.string(),
+      }),
+    )
+    .optional(),
+  input: zod
+    .object({
+      metafield_identifiers: zod
+        .object({
+          namespace: zod.string(),
+          key: zod.string(),
+        })
+        .optional(),
+    })
+    .optional(),
+})
+  .refine((schema) => !schema.supports_3ds || schema.confirmation_callback_url, {
+    message: 'Property required when supports_3ds is true',
+    path: ['confirmation_callback_url'],
+  })
+  .refine((schema) => schema.supports_installments === schema.supports_deferred_payments, {
+    message: 'supports_installments and supports_deferred_payments must be the same',
+  })
+
+export async function creditCardPaymentsAppExtensionDeployConfig(
+  config: CreditCardPaymentsAppExtensionConfigType,
+): Promise<{[key: string]: unknown} | undefined> {
+  return {
+    target: config.targeting[0]!.target,
+    api_version: config.api_version,
+    start_payment_session_url: config.payment_session_url,
+    start_refund_session_url: config.refund_session_url,
+    start_capture_session_url: config.capture_session_url,
+    start_void_session_url: config.void_session_url,
+    confirmation_callback_url: config.confirmation_callback_url,
+    merchant_label: config.merchant_label,
+    supported_countries: config.supported_countries,
+    supported_payment_methods: config.supported_payment_methods,
+    test_mode_available: config.test_mode_available,
+    default_buyer_label: config.buyer_label,
+    buyer_label_to_locale: config.buyer_label_translations,
```

Comment:
`buyer_label_to_locale` and `default_buyer_label` and be removed here as the fields are deprecated: 
https://github.com/Shopify/shopify/blob/main/components/payment_processing/payments_partners/app/models/payments_partners/payments_app_extensions/credit_card_payments_app_extension.rb#L59

Reply:
Good catch!